src/js/components/Button/index.d.ts

```diff
@@ -38,6 +38,7 @@ export interface ButtonProps {
   href?: string;
   target?: '_self' | '_blank' | '_parent' | '_top' | string;
   icon?: JSX.Element;
+  kind?: string | ButtonThemeType;
```

Comment:
After some discussion with @halocline, we've decided to simplify the type definition to just "string" for now since this is the typescript use case we are aiming to solve.

Internally in grommet, there are a few places where we assign `kind` to a theme object. However, in order to support this in a typescript project (where someone can define their own custom `kind` in their theme), there would need to be additional enhancements to our Button typing. Given that `kind` is not even a documented prop and just something we've used within grommet-theme-hpe, I think we should consider the added support of a button type a separate enhancement (and at that point want to consider documenting `kind` as well.